### PR TITLE
[Build System: build-script-impl] Ignore swift-api-dump.py and swift_build_sdk_interfaces.py when running dsymutil on executables and shared libraries.

### PR DESCRIPTION
**Pull request by Rostepher**
Related to #26644.

### CODE REVIEW

**Comment by Rostepher** [comment]
@swift-ci please test

File: utils/build-script-impl [hunk 1]
```diff
@@ -3881,6 +3881,8 @@ for host in "${ALL_HOSTS[@]}"; do
             (cd "${INSTALL_SYMROOT}" &&
              find ./"${CURRENT_PREFIX}" -perm -0111 -type f -print | \
                grep -v crashlog.py | \
+               grep -v swift-api-dump.py | \
+               grep -v swift_build_sdk_interfaces.py | \
                grep -v symbolication.py | \
```
Comment (jrose-apple) [inline utils/build-script-impl]: Should we just `grep -v .py` and be done with it?

Comment (Rostepher) [reply utils/build-script-impl]: I can make that change if you like.

**Comment by Rostepher** [comment]
@swift-ci please smoke test

**Review by shahmishal — APPROVED** [review]

**Pull request merged**